pallets/parachain-staking/src/mock.rs

```diff
@@ -121,132 +121,85 @@ impl Config for Test {
+	collators: Vec<(AccountId, Balance)>,
+	// [nominator, collator, nomination_amount]
+	nominators: Vec<(AccountId, AccountId, Balance)>,
 }
 
-pub(crate) fn two_collators_four_nominators() -> sp_io::TestExternalities {
-	genesis(
-		vec![
-			(1, 1000),
-			(2, 300),
-			(3, 100),
-			(4, 100),
-			(5, 100),
-			(6, 100),
-			(7, 100),
-			(8, 9),
-			(9, 4),
-		],
-		vec![
-			// collators
-			(1, None, 500),
-			(2, None, 200),
-			// nominators
-			(3, Some(1), 100),
-			(4, Some(1), 100),
-			(5, Some(2), 100),
-			(6, Some(2), 100),
-		],
-	)
+impl Default for ExtBuilder {
+	fn default() -> ExtBuilder {
+		ExtBuilder {
+			balances: vec![],
+			nominators: vec![],
+			collators: vec![],
+		}
+	}
 }
 
-pub(crate) fn five_collators_no_nominators() -> sp_io::TestExternalities {
-	genesis(
-		vec![
-			(1, 1000),
-			(2, 1000),
-			(3, 1000),
-			(4, 1000),
-			(5, 1000),
-			(6, 1000),
-			(7, 33),
-			(8, 33),
-			(9, 33),
-		],
-		vec![
-			// collators
-			(1, None, 100),
-			(2, None, 90),
-			(3, None, 80),
-			(4, None, 70),
-			(5, None, 60),
-			(6, None, 50),
-		],
-	)
-}
+impl ExtBuilder {
+	pub(crate) fn with_balances(mut self, balances: Vec<(AccountId, Balance)>) -> Self {
+		self.balances = balances;
+		self
+	}
 
-pub(crate) fn five_collators_five_nominators() -> sp_io::TestExternalities {
-	genesis(
-		vec![
-			(1, 100),
-			(2, 100),
-			(3, 100),
-			(4, 100),
-			(5, 100),
-			(6, 100),
-			(7, 100),
-			(8, 100),
-			(9, 100),
-			(10, 100),
-		],
-		vec![
-			// collators
-			(1, None, 20),
-			(2, None, 20),
-			(3, None, 20),
-			(4, None, 20),
-			(5, None, 10),
-			// nominators
-			(6, Some(1), 10),
-			(7, Some(1), 10),
-			(8, Some(2), 10),
-			(9, Some(2), 10),
-			(10, Some(1), 10),
-		],
-	)
-}
+	pub(crate) fn with_collators(mut self, collators: Vec<(AccountId, Balance)>) -> Self {
+		self.collators = collators;
+		self
+	}
 
-pub(crate) fn one_collator_two_nominators() -> sp_io::TestExternalities {
-	genesis(
-		vec![(1, 100), (2, 100), (3, 100), (4, 100), (5, 100), (6, 100)],
-		vec![
-			// collators
-			(1, None, 20),
-			// nominators
-			(2, Some(1), 10),
-			(3, Some(1), 10),
-		],
-	)
+	pub(crate) fn with_nominators(
+		mut self,
+		nominators: Vec<(AccountId, AccountId, Balance)>,
+	) -> Self {
+		self.nominators = nominators;
+		self
+	}
+
+	pub(crate) fn build(self) -> sp_io::TestExternalities {
+		let mut t = frame_system::GenesisConfig::default()
+			.build_storage::<Test>()
+			.unwrap();
+
+		pallet_balances::GenesisConfig::<Test> {
+			balances: self.balances,
+		}
+		.assimilate_storage(&mut t)
+		.unwrap();
+
+		let mut stakers: Vec<(AccountId, Option<AccountId>, Balance)> = Vec::new();
+		for collator in self.collators {
+			stakers.push((collator.0, None, collator.1));
+		}
+		for nominator in self.nominators {
+			stakers.push((nominator.0, Some(nominator.1), nominator.2));
+		}
+		stake::GenesisConfig::<Test> {
+			stakers,
+			inflation_config: InflationInfo {
+				expect: Range {
```

Comment:
I would be nice if the inflationInfo could also be set up as nominators and collators. Other than that this looks great!

Reply:
done